libbeat/processors/annotate/kubernetes/podwatcher.go

```diff
@@ -0,0 +1,298 @@
+package kubernetes
+
+import (
+	"github.com/elastic/beats/libbeat/logp"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
```

Comment:
@ruflin , im in agreement with @exekias, the kube client is a quite advanced client implementation. its not worth the effort to write our own client. we can have these vendored local to the kubernetes plugin.